armi/reactor/blueprints/tests/test_blockBlueprints.py

```diff
@@ -317,6 +318,40 @@ def test_explicitFlags(self):
         self.assertTrue(a1.hasFlags(Flags.FUEL, exact=True))
         self.assertTrue(a2.hasFlags(Flags.FUEL | Flags.TEST, exact=True))
 
+    def test_densityConsistentWithComponentConstructor(self):
+        # when comparing to 3D density, the comparison is not quite correct.
+        # We need a bigger delta, this will be investigated/fixed in another PR
+        biggerDelta = 0.001  # g/cc
+        a1 = self.blueprints.assemDesigns.bySpecifier["IC"].construct(
+            self.cs, self.blueprints
+        )
+        fuelBlock = a1[0]
+        clad = fuelBlock.getComponent(Flags.CLAD)
+
+        # now construct clad programmatically like in test_Blocks
+        programmaticBlock = test_blocks.buildSimpleFuelBlock()
+        programaticClad = programmaticBlock.getComponent(Flags.CLAD)
+        self.assertAlmostEqual(
+            clad.getMassDensity(),
+            clad.material.density3(Tc=clad.temperatureInC),
+            delta=biggerDelta,
+        )
+        # This should be equal, but block construction calls applyHotHeightDensityReduction
+        # while programmatic construction allows components to exist in a state where
+        # their density is not consistent with material density.
+        self.assertNotAlmostEqual(
```

Comment:
We probably want to fix this. it is documented in https://github.com/terrapower/armi/issues/819

Reply:
I have been waffling back-and-forth on this.

Adding a test to the repo that passes, but shouldn't seems strange.

But sure, it serves a purpose. I'm on board for that.

Can we add a line: `# TODO: This test passes, but shouldn't` or something, just to ease my nerves?